src/logic/exceptions/CodedException.ts

```diff
@@ -1,23 +1,28 @@
 import * as Sentry from '@sentry/react'
+import { CaptureContext } from '@sentry/types'
 import ErrorCodes from './registry'
 import { IS_PRODUCTION } from 'src/utils/constants'
 
 export class CodedException extends Error {
-  public readonly message: string
   public readonly code: number
+  // the context allows to enrich events, for the list of allowed context keys/data, please check the type or go to
+  // https://docs.sentry.io/platforms/javascript/enriching-events/context/
+  // The context is not searchable, that means its goal is just to provide additional data for the error
+  public readonly sentryContext?: CaptureContext
 
-  constructor(content: ErrorCodes, extraMessage?: string) {
+  constructor(content: ErrorCodes, extraMessage?: string, context?: CaptureContext) {
```

Comment:
So `contexts` can be basically anything? Maybe we can enforce some internal type to keep the events uniform?

I would get rid of `extraMessage` then, so that we always pass extra info in the context.

Reply:
I did that first too, but then I discovered that only tags are searchable from the context and though it's not something we want

Citing the description:
> I left functionality for concatenating messages (extraMessage) because from context fields only the tag field is searchable. Thought it would be handy in case we get a bunch of the same errors and we're able to search and remove them.

Reply:
If you think that's fine, I can remove it

Reply:
> So contexts can be basically anything?

Not really, each of those has its clear purpose except maybe context/extras:
```
export interface ScopeContext {
    user: User; // self-explanatory
    level: Severity; // self-explanatory
    extra: Extras;  // key-value (any) pairs
    contexts: Contexts; // key-value (any) pairs
    tags: {
        [key: string]: Primitive; // SEARCHABLE key-primitive pairs 
    };
    fingerprint: string[]; // self-explanatory
}
```

I'd we don't really need user/fingerprinting fields and we can remove them via our own type. The scope is also editable globally via `sentry.setTag` for example and that means we'll have more places where we need to redeclare things :) I'd keep it the way it is right now

Reply:
For me it's a question of grouping, not necessarily searching. I think it's better to have fewer possible types of errors on Sentry grouped by the main error message.
If we need to investigate why some particular error is happening or spiking, we'll just dig into them and see the contexts.

Reply:
> Not really, each of those has its clear purpose except maybe context/extras:

Sorry, I meant the actual `contexts` field.

> I'd keep it the way it is right now
Yeah, let's see if a usage pattern emerges, then we might encode it into our own strict type for this field.

Reply:
> Sorry, I meant the actual contexts field.

I don't think that enforcing it via a type would make sense. For example, the Safe Apps part of the app needs a vastly different context than any other part.

> For me it's a question of grouping, not necessarily searching. I think it's better to have fewer possible types of errors on Sentry grouped by the main error message.
If we need to investigate why some particular error is happening or spiking, we'll just dig into them and see the contexts.

And for me, it was a question of discoverability. Let me tell you a hypothetical situation I had in mind when leaving `extraMessage`:

Suddenly there is a server outage and we get hundreds of `901: Error processing Safe Apps request`. They don't mean anything to us because nothing needs to be fixed on our side and we want to remove them. If the message doesn't contain any extra info we need to go through errors 1 by 1 and check the additional context to see what the underlying error was.  If the name contains an extra message so it would be `901: Error processing Safe Apps: Server responded with 404` request, we can just search for this message and remove all the errors in bulk. I see the second option as a better one. I'm not a sentry expert, If sentry has a mechanism for not creating 1000 same errors then we can remove it.

Reply:
> For example, the Safe Apps part of the app needs a vastly different context than any other part.

These vastly different types can still be described as a union.
```
type Context = SafeAppContext | ReactAppContext | SomeOtherContext
```

> hypothetical situation I had in mind when leaving extraMessage

Yes, makes sense. So we still group, but based on the main error + the original error. 👍